### DISCUSSION TITLE
Delete links no more referred by other issues and pull-requests

### DISCUSSION THREAD

**Discussion by peaceiris** [community/community#4350] [Projects and Issues]
When we mention another issue (or pull-request) from an issue (or pull-request), a link will be created as follows.

![Screen Shot 2021-06-27 at 11 39 42](https://user-images.githubusercontent.com/30958501/123531077-62c4b400-d73c-11eb-9917-d405742f784f.png)

But, the link never is deleted when other issues (or pull-requests) stop to reference it. Reflecting the latest linking is better to reduce noise.

**Comment by azenMatt** [comment]
Hi @peaceiris - this feels like a bug... i'l report it to the team 👍

**Reply by ferdnyc** [reply]
Just to add a little support, it's a particular nuisance when you had _no intention_ of linking the two issues. For example... (these are all real scenarios I've encountered in the wild):
1. You mentioned the wrong issue/PR by number, and later corrected it
2. You pasted a URL to the issue into a comment on a different issue at an unrelated repo, and had no intention of creating noise for the developers of the linked-to project
3. You pasted in some text content that included lots of numbers preceded by hash symbols (common in certain types of log output or debugger output) without fencing it inside a code block, causing it to "mention" dozens of random issues (usually with low, consecutive numbers)
4. You're in the habit of writing things like "number one" (when discussing lists of options, or the contents of a prior ordered list) as `#1` in haste, forgetting that GitHub is going to link that to issue 1 in whatever repo you're commenting in. (That one's me. I do it at least once a week.)

The unretractable mention is like getting your mistake marked down on that "permanent record" we all heard about as kids, the one that isn't supposed to actually exist. :roll_eyes:
